package/MDAnalysis/analysis/align.py

```diff
@@ -504,6 +504,14 @@ def alignto(mobile, reference, select=None, weights=None,
     if subselection is None:
         # mobile_atoms is Universe
         mobile_atoms = mobile.universe.atoms
+        if mobile.universe == reference.universe:
+            # raise a warning when passing atom groups from same Universe
+            wmsg = ("The reference and mobile selections are atom groups "
+                   "from the same universe, the reference selection will " 
+                   "be transformed. If it is not the behavior you intend, " 
+                   "try setting the subselection argument to 'all'.This "
+                   "will become the default behavior in version 3.0.0.")
+            warnings.warn(wmsg, DeprecationWarning)
```

Comment:
Could you add a `.. deprecated` entry in the docstring above?

Reply:
Please do remember to address this and my other comment https://github.com/MDAnalysis/mdanalysis/pull/3562/files#r834027990

Reply:
Please address this before I re-review

Reply:
I must tell you I'm not really sure what to do here or how to address it.
When writing `subselection = 'all' is deprecated in 2.1.0 but it` in line 513 am I not adding `.. deprecated` entry?
Sorry for this :(.

Reply:
@JoseAntonioSiguenza when doing a change like this we want to make really really super sure that people see it, so we put notices in three ways:

* in the change log (which you've done)
* as a warning (as you do here)
* and in the function or class documentation.

Irfan's asking to do the third one, as you can see higher up in the function:

https://github.com/MDAnalysis/mdanalysis/blob/a13866b4843da1b1b351e83c7967250a4b1688d7/package/MDAnalysis/analysis/align.py#L457-L475 It goes newest to oldest, so your new entry should go on top of the 1.0.0 `versionchanged`.

However, I don't understand what is being deprecated here:

> "`subselection = 'all' is deprecated in 2.1.0 but it "
> "will become the default behavior in version 3.0.0."

Firstly, we have already released version 2.1, and you have not changed the default from `subselection=None`. Did you mean 2.2? Secondly, I think it is `subselection=None`, and the logic that follows, that is becoming deprecated.

Reply:
Could you please also explain in the error message how the reference selection is transformed, and how setting `subselection="all"` changes that behaviour?

Reply:
Now I see, thanks! Yes, it was the 2.2 version and I intend to inform the user that the setting `subselection=None` is becoming deprecated.

So, the documentation update should look something like this?

.. versionchanged:: 2.2.0
    When working with `AtomGroup` from the same `Universe`, `subselection="all"` should be considered

Reply:
In this case you should be issuing a deprecation notice, i.e.

```
... deprecated:: 2.2.0
    In version 3.0.0, when the reference and mobile selections are atom groups from the same 
    universe and ``subselection`` is ``None``, the reference selection will no longer be transformed. 
    This will result in the same behaviour what is currently obtained in the above case but with 
    ``subselection`` set to 'all'.
```

Reply:
Also I agree with @lilyminium - your current warning message does not make sense on re-reading, `subselection = 'all'` is not deprecated, instead the default behaviour for `subselection = None` for this specific case will change to match that of what is currently obtained using `subselection = 'all'`

Reply:
Exact. When the warning is passed I intend to refer to the behavior you are stating. I thought it was a `DeprecationWarning` message that should be passed, do you have any suggestions on the type of message to appear? Should it be just a `Warning` message?

Reply:
I'm not sure I understand what you mean here @JoseAntonioSiguenza.

Do you mean that it's not strictly a "deprecation" in its sense (since we're not removing anything).

If so we could opt for a `.. warning` (containing something akin to this notice https://github.com/MDAnalysis/mdanalysis/pull/3562/files#r838509418) and a `FutureWarning` (which rewords this sufficiently to deal with this https://github.com/MDAnalysis/mdanalysis/pull/3562/files#r838511631)

Reply:
Exactly, I'm changing this for a `.. warning` instead of `.. deprecation`. Thanks for the feedback! :)

Reply:
I've made the requested changes at the last commit. It should be good now.